internal/mbsc/mbsc.go

```diff
@@ -46,30 +47,39 @@ func (m *mbsc) Get(ctx context.Context, name, namespace string) (*kmmv1beta1.Mod
 	return &mbsc, nil
 }
 
+/*
 func (m *mbsc) CreateOrPatch(ctx context.Context,
+
 	name string,
 	namespace string,
 	moduleImageSpec *kmmv1beta1.ModuleImageSpec,
 	imageRepoSecret *v1.LocalObjectReference,
 	owner metav1.Object) error {
+*/
+func (m *mbsc) CreateOrPatch(ctx context.Context, micObj *kmmv1beta1.ModuleImagesConfig,
+	moduleImageSpec *kmmv1beta1.ModuleImageSpec, action kmmv1beta1.BuildOrSignAction) error {
 	mbscObj := &kmmv1beta1.ModuleBuildSignConfig{
-		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
+		ObjectMeta: metav1.ObjectMeta{Name: micObj.Name, Namespace: micObj.Namespace},
 	}
 
 	_, err := controllerutil.CreateOrPatch(ctx, m.client, mbscObj, func() error {
-		setModuleImageSpec(mbscObj, moduleImageSpec)
-		mbscObj.Spec.ImageRepoSecret = imageRepoSecret
-		return controllerutil.SetOwnerReference(owner, mbscObj, m.scheme)
+		setModuleImageSpec(mbscObj, moduleImageSpec, action)
+		mbscObj.Spec.ImageRepoSecret = micObj.Spec.ImageRepoSecret
+		return controllerutil.SetOwnerReference(micObj, mbscObj, m.scheme)
 	})
 	return err
 }
 
-func setModuleImageSpec(mbscObj *kmmv1beta1.ModuleBuildSignConfig, moduleImageSpec *kmmv1beta1.ModuleImageSpec) {
+func setModuleImageSpec(mbscObj *kmmv1beta1.ModuleBuildSignConfig, moduleImageSpec *kmmv1beta1.ModuleImageSpec, action kmmv1beta1.BuildOrSignAction) {
+	newEntry := kmmv1beta1.ModuleBuildSignSpec{
```

Comment:
```suggestion
	mbsSpec := kmmv1beta1.ModuleBuildSignSpec{
```
nit: just because it doesn't have to be a new entry but also an updated existing entry.

Reply:
fixed